### PR TITLE
Add ContainerTracker for tracking deployed docker containers

### PR DESCRIPTION
**Pull request by nkubala**
**Related**: https://github.com/GoogleContainerTools/skaffold/issues/6107 

**Description**

This change adds a `ContainerTracker` object, which tracks all containers deployed to the local Docker daemon by Skaffold. This object's purpose is analogous to the `PodSelector`, which is used for all Kubernetes-based `Deployer` sub-components. 

The `ContainerTracker` will be passed to all future implementations of these sub-components in the Docker deployer, and will inform those components which resources to retrieve and operate on. It will also help the `ColorPicker` select the proper colors for individual container logs (when implemented).

**Follow-up Work (remove if N/A)**

Use `ContainerTracker` to implement `Deployer` sub-components for the Docker deployer.

### CODE REVIEW

File: pkg/skaffold/docker/tracker/tracker.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		containers:          make(map[string]bool),
+		containerToArtifact: make(map[string]graph.Artifact),
+		notifier:            make(chan string, 1),
+	}
+}
+
+// GetNotifier returns the notifier channel for this tracker.
+// Used by the log streamer to be notified when a new container is
+// added to the tracker.
+func (t *ContainerTracker) GetNotifier() chan string {
```
Comment (MarlonGamez) [inline pkg/skaffold/docker/tracker/tracker.go:47]: nit: just call this `Notifier()`

File: pkg/skaffold/docker/tracker/tracker.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	return t.notifier
+}
+
+// ImageForContainer maps a container id to the image tag it was created from.
+// Used by the ColorPicker to maintain consistency between images and colors.
+func (t *ContainerTracker) ArtifactForContainer(id string) graph.Artifact {
+	return t.containerToArtifact[id]
+}
+
+// DeployedContainerForImage returns a the ID of a deployed container create from
```
Comment (MarlonGamez) [inline pkg/skaffold/docker/tracker/tracker.go:57]: nit:
```suggestion
// DeployedContainerForImage returns a the ID of a deployed container created from
```

**Review by MarlonGamez — APPROVED** [review]
have a couple nits but lgtm

**Comment by nkubala** [comment]
CI is failing because of https://github.com/GoogleContainerTools/skaffold/issues/6262. self merging

**Pull request merged**